### DISCUSSION TITLE
Payload data formats for exchange APIs

### DISCUSSION THREAD

**Discussion by ghost** [spdci/.github#7] [General]
Payload data formats for exchange APIs

**Comment by vvujjini** [comment]
I'm requesting [this](https://github.com/spdci/standards) API specs to be internally reviewed and opened up for public comments to adopt as standards. The repo has been forked out of g2p connect spec repo and further improved for accommodating multiple registries and data format/codes.

Below are few key features of these specs:
1. Keep the api code [modular](https://redocly.com/docs/resources/multi-file-definitions/) so that each end point & schema object definitions are independently managed without breaking other parts.
2. Easy readability for the contributors to suggest or make changes to APIs specs.
3. Allow versioning of minor and patch to schema objects while ensuring backward compatibility to the api's with in a major version.
4. Enable extensibility to inject custom country codes in the API design. For e.g. DCI codes, FHIR, OpenID, etc.,  codes, can be easily added in a separate [folder](https://github.com/spdci/standards/tree/draft/src/extensions) and referenced in required request/response payload objects.
5. This specs shall also support adding other functional registries to the same API end points like farmer registry, disability etc., and future proofing to enable one harmonised API design/structure for system to system access of foundational or functional registries.  
6. All of the capabilities (except for get subscriptions and unsubscribe end points) discussed till date by various committee members are retained in this refactored code base. 

I do believe this can further be improved as more use cases are discussed by the community. I propose these draft **standards** branch be restricted to a small group to review and accept pull requests to strengthen the quality process. IMO, specs should also be treated as code and bring in the best coding/release mgmt practices to ensure backward compatibility and integrity of the process. 

I'm hoping these specs will be reviewed internally to ensure all critical feedback received in past weeks is present before opening up for public review comments.

(todo: Get subscriptions and unsubscribe features to be added)
